flow/core/kernel/scenario/aimsun.py

```diff
@@ -73,39 +78,80 @@ def generate_network(self, scenario):
-                               'flow/utils/aimsun/generate.py')
+        template_path = scenario.net_params.template
+        if template_path is None:
+            script_path = osp.join(config.PROJECT_PATH,
+                                   'flow/utils/aimsun/generate.py')
+        else:
+            script_path = osp.join(config.PROJECT_PATH,
+                                   'flow/utils/aimsun/load.py')
+            file_path = osp.join(config.PROJECT_PATH,
+                                 'flow/utils/aimsun/aimsun_template_path')
+            with open(file_path, 'w') as f:
+                f.write(template_path)
 
         # start the aimsun process
         aimsun_call = [aimsun_path, "-script", script_path]
         self.aimsun_proc = subprocess.Popen(aimsun_call)
 
         # merge types into edges
         if scenario.net_params.osm_path is None:
-            for i in range(len(scenario.edges)):
-                if 'type' in scenario.edges[i]:
-                    for typ in scenario.types:
-                        if typ['id'] == scenario.edges[i]['type']:
-                            new_dict = deepcopy(typ)
-                            new_dict.pop("id")
-                            scenario.edges[i].update(new_dict)
-                            break
-
-            self._edges = {}
-            for edge in deepcopy(scenario.edges):
-                edge_name = edge['id']
-                self._edges[edge_name] = {}
-                del edge['id']
-                self._edges[edge_name] = edge
+            if scenario.net_params.template is None:
+                for i in range(len(scenario.edges)):
+                    if 'type' in scenario.edges[i]:
+                        for typ in scenario.types:
+                            if typ['id'] == scenario.edges[i]['type']:
+                                new_dict = deepcopy(typ)
+                                new_dict.pop("id")
+                                scenario.edges[i].update(new_dict)
+                                break
+
+                self._edges = {}
+                for edge in deepcopy(scenario.edges):
+                    edge_name = edge['id']
+                    self._edges[edge_name] = {}
+                    del edge['id']
+                    self._edges[edge_name] = edge
+
+                # list of edges and internal links (junctions)
+                self._edge_list = [
+                    edge_id for edge_id in self._edges.keys()
+                    if edge_id[0] != ':'
+                ]
+                self._junction_list = list(
+                    set(self._edges.keys()) - set(self._edge_list))
+
+            else:
+                # load scenario from template
+                scenario_file = "flow/core/kernel/scenario/scenario_data.json"
+                filepath = os.path.join(config.PROJECT_PATH, scenario_file)
+
+                while not os.path.exists(filepath):
+                    time.sleep(0.5)
```

Comment:
why do we want 0.5 s here? does the file take that long to create? maybe we can try something smaller, like 0.1s

Reply:
fixed in b524edc